Scripts/Runtime/Entities/Util/WorldUtil.cs

```diff
@@ -0,0 +1,230 @@
+    /// </summary>
+    internal class PostUpdate
+    {
+        /// <summary>
+        /// Provides a <see cref="PlayerLoopSystem" /> instance to to identify this phase.
+        /// </summary>
+        public static PlayerLoopSystem PlayerLoopSystem
+        {
+            get => new PlayerLoopSystem()
+            {
+                type = typeof(PostUpdate)
+            };
+        }
+    }
+
+    /// <summary>
+    /// A <see cref="PlayerLoop"/> phase inserted immediately after <see cref="Initialization"/>.
+    /// Add via <see cref="WorldUtil.AddCustomPhasesToCurrentPlayerLoop" />.
+    /// </summary>
+    internal class PostInitialization
+    {
+        /// <summary>
+        /// Provides a <see cref="PlayerLoopSystem" /> instance to to identify this phase.
+        /// </summary>
+        public static PlayerLoopSystem PlayerLoopSystem
+        {
+            get => new PlayerLoopSystem()
+            {
+                type = typeof(PostInitialization)
+            };
+        }
+    }
+
+    /// <summary>
+    /// A collection of utilities to manipulate and augments <see cref="World"/>s.
+    /// The Anvil compliment to <see cref="ScriptBehaviourUpdateOrder"/>.
+    /// </summary>
+    public static class WorldUtil
+    {
+        // Not need to reset between play sessions because PlayerLoop systems are stateless and 
+        // persist between sessions when domain reloading is disabled.
+        private static bool s_AreCustomPlayerLoopPhasesAdded = false;
+
+        /// <summary>
+        /// Add custom phases to the <see cref="PlayerLoop"/>.
+        /// <see cref="PostInitialization"/> and <see cref="PostUpdate"/>.
+        /// </summary>
+        public static void AddCustomPhasesToCurrentPlayerLoop()
+        {
+            if (s_AreCustomPlayerLoopPhasesAdded)
+            {
+                return;
+            }
+
+            // Create new top level player loop phases
+            PlayerLoopSystem playerLoop = PlayerLoop.GetCurrentPlayerLoop();
+            List<PlayerLoopSystem> topLevelPhases = playerLoop.subSystemList.ToList();
+
+            int initializationPhaseIndex = topLevelPhases.FindIndex((phase) => phase.type == typeof(Initialization));
+            Debug.Assert(initializationPhaseIndex != -1, "Initialization phase not found");
+            Debug.Assert(!topLevelPhases.Any((phase) => phase.type == typeof(PostInitialization)), $"{nameof(PostInitialization)} phase already added");
+            topLevelPhases.Insert(initializationPhaseIndex + 1, PostInitialization.PlayerLoopSystem);
+
+            int updatePhaseIndex = topLevelPhases.FindIndex((phase) => phase.type == typeof(Update));
+            Debug.Assert(updatePhaseIndex != -1, "Update phase not found");
+            Debug.Assert(!topLevelPhases.Any((phase) => phase.type == typeof(PostUpdate)), $"{nameof(PostUpdate)} phase already added");
+            topLevelPhases.Insert(updatePhaseIndex + 1, PostUpdate.PlayerLoopSystem);
```

Comment:
Seems like this and lines 89 to 92 could be made into a function where you pass in the types?

Reply:
Sort of, In this case they're both "AddAfter X" operations but there's any number of ways to decide where a phase lands. It didn't feel necessary to create a method just because these two phases happened to get positioned in the same way.

Reply:
Sounds good